lib/ansible/modules/cloud/dimensiondata/dimensiondata_firewall.py

```diff
@@ -0,0 +1,509 @@
+    return DimensionDataFirewallAddress(any_ip, ip_address, ip_prefix_size,
+                                        start_port, end_port, None, None)
+
+
+def sync_firewall_rule_state(module, driver, fw_rule, state):
+    if state == 'disabled' and fw_rule.enabled != 'false':
+        enabled = False
+    elif state != 'disabled' and fw_rule.enabled == 'true':
+        return {'success': True, 'result': 'nochange'}
+    elif state != 'disabled' and fw_rule.enabled != 'true':
+        enabled = True
+
+    try:
+        # State doesnt match, so set state.
+        driver.ex_set_firewall_rule_state(fw_rule, enabled)
+        fw_rule.enabled = 'false' if state == 'disabled' else 'true'
+        module.exit_json(changed=True,
+                         msg="Firewall rule %s state changed." %
+                         fw_rule.name, firewall_rule=rule_obj_to_dict(fw_rule))
+    except DimensionDataAPIException:
+        e = get_exception()
+        module.fail_json(msg="Firewall rule state change failed with: %s" % e)
+
+
+def rule_obj_to_dict(rule):
+    rule_dict = dict(id=rule.id, name=rule.name, action=rule.action)
+    if isinstance(rule.location, NodeLocation):
+        rule_dict['location'] = dict(id=rule.location.id,
+                                     name=rule.location.name,
+                                     country=rule.location.country)
+    else:
+        rule_dict['location'] = rule.location
+    rule_dict['status'] = rule.status
+    rule_dict['ip_version'] = rule.ip_version
+    rule_dict['protocol'] = rule.protocol
+    rule_dict['enabled'] = rule.enabled
+
+    return rule_dict
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            region=dict(default='na'),
+            mcp_user=dict(required=False, type='str'),
+            mcp_password=dict(required=False, type='str'),
+            location=dict(required=True, type='str'),
+            network_domain=dict(required=True, type='str'),
+            name=dict(required=True, type='str'),
+            state=dict(default='present', choices=['present', 'absent', 'enabled', 'disabled']),
+            action=dict(default='ACCEPT_DECISIVELY', choices=['ACCEPT_DECISIVELY', 'DROP']),
+            ip_version=dict(default='IPv4', choices=['IPv4', 'IPv6']),
+            protocol=dict(default='TCP', choices=['IP', 'ICMP', 'TCP', 'UDP']),
+            source=dict(required=False, default='ANY', type='str'),
+            source_port=dict(required=False, default='ANY', type='str'),
+            destination=dict(required=False, default='ANY', type='str'),
+            destination_port=dict(required=False, default='ANY', type='str'),
+            position=dict(default='LAST', choices=['FIRST', 'LAST', 'BEFORE', 'AFTER']),
+            relative_to_rule=dict(required=False, default=None, type='str'),
+            verify_ssl_cert=dict(required=False, default=True, type='bool')
+        ),
+        required_together=[
+            ['mcp_user', 'mcp_password']
```

Comment:
Looks good. You can test this by specifying one but not the other in a task and ensuring you get a sensible error